CRM/MembershipExtras/Hook/Alter/CalculatedMembershipStatus.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+/**
+ * Calculates membership status taking into account in_arrears and not_arrears
+ * start and end events.
+ */
+class CRM_MembershipExtras_Hook_Alter_CalculatedMembershipStatus {
+
+  private $membership;
+  private $calculationArguments;
+  private static $memberShipStatuses = array();
+  private static $contributionStatusValueMap = array();
+
+  public function __construct() {
+    if (count(self::$memberShipStatuses) == 0) {
+      $membershipStatuses = civicrm_api3('MembershipStatus', 'get', [
+        'sequential' => 1,
+        'options' => ['sort' => 'weight ASC', 'limit' => 0],
+      ]);
+      self::$memberShipStatuses = $membershipStatuses['values'];
+    }
+
+    if (count(self::$contributionStatusValueMap) == 0) {
+      $contributionStatuses = civicrm_api3('OptionValue', 'get', array(
+        'option_group_id' => "contribution_status",
+      ));
+
+      foreach ($contributionStatuses['values'] as $currentStatus) {
+        self::$contributionStatusValueMap[$currentStatus['name']] = $currentStatus['value'];
+      }
+    }
+  }
+
+  /**
+   * This hook is called when membership status is being calculated.
+   *
+   * @param array $calculatedStatus
+   *   Membership status details as determined by CiviCRM core
+   * @param array $arguments
+   *   Arguments passed in to calculate date
+   * @param array $membership
+   *   Membership details from the calling function
+   */
+  public function alterMembershipStatus(&$calculatedStatus, $arguments, $membership) {
+    $this->membership = $membership;
+    $this->calculationArguments = $arguments;
+
+    foreach (self::$memberShipStatuses as $currentStatus) {
+      $startEventIsArrearsRelated = stripos($currentStatus['start_event'], 'arrears') !== false;

```

Comment:
Yeah, I wanted to capture both 'In arrears' and 'Not in arrears' in a single instruction, and not worry about upper and lower case letters.

Reply:
but civicrm API return the machine name for the start and end event, so in this case it can be either in_arrears or not_arrears since these are the machine names u used, correct ? 
 
If I am not missing anything, it will be more robust to use the exact machine name rather than stripos.